Sources/WalletConnectPush/UnregisterService.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+
+actor UnregisterService {
+    private let httpClient: HTTPClient
+    private let projectId: String
+    private let logger: ConsoleLogging
+    private let environment: APNSEnvironment
+    private let pushAuthenticator: PushAuthenticating
+    private let clientIdStorage: ClientIdStoring
+    private let pushHost: String
+
+    init(httpClient: HTTPClient,
+         projectId: String,
+         clientIdStorage: ClientIdStoring,
+         pushAuthenticator: PushAuthenticating,
+         logger: ConsoleLogging,
+         pushHost: String,
+         environment: APNSEnvironment) {
+        self.httpClient = httpClient
+        self.clientIdStorage = clientIdStorage
+        self.pushAuthenticator = pushAuthenticator
+        self.projectId = projectId
+        self.logger = logger
+        self.pushHost = pushHost
+        self.environment = environment
+    }
+
+    func unregister() async throws {
+            let pushAuthToken = try pushAuthenticator.createAuthToken()
+            let clientId = try clientIdStorage.getClientId()
+
+            guard let url = URL(string: "https://\(pushHost)/\(projectId)/clients/\(clientId)") else {
+                logger.error("Invalid URL")
+                return
+            }
+
+            var request = URLRequest(url: url)
+            request.httpMethod = "DELETE"
+            request.addValue("\(pushAuthToken)", forHTTPHeaderField: "Authorization")
+
+            do {
+                let (_, response) = try await URLSession.shared.data(for: request)
+
+                guard let httpResponse = response as? HTTPURLResponse, httpResponse.statusCode == 200 else {
```

Comment:
> 200

Instead of checking for a specific status code, can you check if it is any 2xx status code? We may want to e.g. change to 204 (no content) and all 2xx codes mean success